test/domain/Vault.spec.js

```diff
@@ -1,17 +1,259 @@
-require('chai').should()
+const should = require('chai').should()
+const { assert } = require('chai')
+const sinon = require('sinon')
+const KeyVaultError = require('../../lib/domain/KeyVaultError')
 const Vault = require('../../lib/domain/Vault')
+const vaultSecrets = require('./vault.json')
 
 describe('Vault', function () {
   describe('getSecrets', function () {
-    it('should return the secrets', function () {
+    it('should return a deep clone of all the secrets from the vault', function () {
       // Given
-      const vault = new Vault()
+      const vault = new Vault(vaultSecrets)
 
       // When
       const secrets = vault.getSecrets()
 
       // Then
-      secrets.should.deep.equal([])
+      secrets.length.should.equal(vaultSecrets.length)
+      secrets.should.not.equal(vaultSecrets)
+      secrets[0].should.not.equal(vaultSecrets[0])
+      secrets[1].should.not.equal(vaultSecrets[1])
+      secrets.should.deep.equal(vaultSecrets)
+    })
+    it('should return a deep clone of a part of the secrets from the vault', function () {
+      // Given
+      const vault = new Vault(vaultSecrets)
+
+      // When
+      const secrets = vault.getSecrets(1)
+
+      // Then
+      secrets.length.should.equal(1)
+      secrets.should.not.equal(vaultSecrets)
+      secrets[0].should.not.equal(vaultSecrets[0])
+      secrets.should.deep.equal(vaultSecrets.slice(0, 1))
+    })
+  })
+  describe('getSecret', function () {
+    it('should return a deep clone secret from the vault', function () {
```

Comment:
on pourrait utiliser des fonctions fléchées

Reply:
non, car mocha perd le contexte dans ce cas-là :
https://remarkablemark.org/blog/2016/08/20/arrow-functions-in-mocha/